packages/extension/src/content-scripts/events.ts

```diff
@@ -31,16 +31,23 @@ class PushEventDataMsg extends Message<void> {
 
 export function initEvents(router: Router) {
   router.registerMessage(PushEventDataMsg);
-
   router.addHandler("interaction-foreground", (_, msg) => {
```

Comment:
If we type `msg` we can avoid the type cast on line 36.

```suggestion
  router.addHandler("interaction-foreground", (_, msg: PushEventDataMsg) => {
```

Reply:
TSC complains about this since the arguments to `router.addHandler` are typed:
```javascript
addHandler(route: string, handler: Handler): void;
```
```javascript
export declare type Handler = (env: Env, msg: Message<unknown>) => any;
```